Readme.md

```diff
@@ -1,21 +1,309 @@
+│   └── Readme.txt
+├── services
+│   └── dashboard_service
+│       ├── dashboard_service.py
+│       ├── javascripts
+│       │   └── poll.js
+│       ├── launch.sh
+│       ├── requirements.txt
+│       └── scripts
+│           ├── build.py
+│           ├── build.sh
+│           └── ipytail.py
+├── setup-node.sh
+└── tmp
+    └── Readme.txt
+```
+
+There is a single docker container that contains JupyterHub and student servers.
+It is setup with docker compose. A web server configuration (nginx configuration provided)
+should be setup on the host to expose this container over a domain.
+
+The container also contains a [Dashboard service](#dashboard-service) that is a Flask app which
+runs some action when a notebook is saved. For now, this is the [build script][ipytail] that
+updates summaries of notebooks when a student makes a save. The `scripts/` directory stores these.
+The `javascripts/` directory has a polling script that uses the endpoint exposed by dashboard-service
+to allow a developer to perform some action on frontend when a particular event (such as save of a notebook)
+is logged. For example, this can be used to notify the user on the frontend that an update is available
+to the summary page.
+
+Configuration files are kept in `etc/`. Of these `jupyterhub/` and `jupyter/` are for inside the
+docker container and `nginx/` is for the host.
+
+#### Container
+
+The jupyterhub container is configured using `docker compose`. [`docker-compose.yml`](/docker-compose.yml) lists several volume mounts and an expose port.
+The volume mounts are either for sharing configuration with the container or for ease of visibility for the trainer.
+
+#### Dashboard service
+
+Related issue: https://github.com/pipalacademy/pipalhub/issues/6
+
+This is a JupyterHub-managed service, i.e. the related process is started and stopped by JupyterHub.
+We only need to configure it in [`jupyterhub_config.py`](https://github.com/pipalacademy/pipalhub/blob/b10a5fa8817c4f3e594a0bd4f5044c157f5092da/etc/jupyterhub/jupyterhub_config.py#L565-L575),
+with the command that needs to run.
+
+Currently this is a Flask app that is started on port 10101 using environment variables
+configured in `jupyterhub_config.py`. JupyterHub will also create a reverse proxy endpoint
+on its server to this service. So, this dashboard service will be accessible at `https://hostname.pipal.in/services/dashboard`,
+and we won't need some separate nginx configuration for this.
+
+The `launch.sh` script for this installs dependencies needed for it to function with pip (flask, pydantic)
+before starting it. This may change in the future, a possible solution would be to have a single
+`requirements.txt` for the instance that comes with defaults but can be changed by the trainer.
+
+##### /events endpoint
+
+`/events` supports `GET` and `POST` methods.
+
+`GET /events` can also be combined with filters as query params.
+
+These are example requests/responses:
+
+###### Create event:
+
+```json
```

Comment:
not JSON. this creates a weird highlighting effect for non-JSON parts of the block.